docs/developers/README.md

```diff
@@ -627,4 +627,20 @@ cat /tekton/steps/step-unnamed-<step-index>/exitCode
 ```
 
 Or, you can access the step metadata directory via symlink, for example, use `cat /tekton/steps/0/exitCode` for the
-first step in a task.
\ No newline at end of file
+first step in a task.
+
+## TaskRun Use of Pod Termination Messages
+
+Tekton Pipelines uses a `Pod's` [termination
+message](https://kubernetes.io/docs/tasks/debug-application-cluster/determine-reason-pod-failure/)
+to pass data from a Step's container to the Pipelines controller.
+Examples of this data include: the time that execution of the user's
+step began, contents of task results, contents of pipeline resource
+results.
+
+The contents and format of the termination message can change. At time
```

Comment:
thanks a bunch @sbwsg for adding this.

NIT: I would add a little note here saying its not recommended to consume the termination message.